http/src/main/resources/init-resources/startup.sh

```diff
@@ -64,6 +65,34 @@ export WELDER_MEM_LIMIT=$(welderMemLimit)
 export MEM_LIMIT=$(memLimit)
 export USE_GCE_STARTUP_SCRIPT=$(useGceStartupScript)
 
+function failScriptIfError() {
+  if [ $EXIT_CODE -ne 0 ]; then
+    echo "Fail to docker-compose start welder ${EXIT_CODE}. Output is saved to ${JUPYTER_START_USER_SCRIPT_OUTPUT_URI}"
+    retry 3 gsutil -h "x-goog-meta-passed":"false" cp start_output.txt ${JUPYTER_START_USER_SCRIPT_OUTPUT_URI}
+    exit $EXIT_CODE
+  else
+    retry 3 gsutil -h "x-goog-meta-passed":"true" cp start_output.txt ${JUPYTER_START_USER_SCRIPT_OUTPUT_URI}
+  fi
+}
+
+# Overwrite old cert on restart
+SERVER_CRT=$(proxyServerCrt)
+SERVER_KEY=$(proxyServerKey)
+ROOT_CA=$(rootCaPem)
+
+## This helps when we need to rotate certs.
+notAfter=`openssl x509 -enddate -noout -in /certs/jupyter-server.crt` # output should be something like `notAfter=Jul 22 13:09:15 2023 GMT`
+
+## If cert is old, then pull latest certs. Update date if we need to rotate cert again
+if [[ "$notAfter" != *"notAfter=Jul 22"* ]] ; then
```

Comment:
Parsing dates seems a little error-prone.. wonder if something like this would be better: https://stackoverflow.com/questions/21297853/how-to-determine-ssl-cert-expiration-date-from-a-pem-encoded-certificate

E.g.
```
EXPIRED=0
openssl x509 -checkend <3-months-in-sections> -noout -in /certs/jupyter-server.crt || EXPIRED=$?
if [ $EXPIRED -ne 0 ]; then
  # copy the cert
fi
```

Reply:
I can...but this means we won't be able to validate things in dev...since dev cert won't expire for another almost 3 years (the one you created in June)...

Reply:
Hm yeah that's true..

Reply:
Do all the new certs in all envs expire on July 22, 2023?

Reply:
yep..I created them on the same date...